### DISCUSSION TITLE
Global search in Database

### DISCUSSION THREAD

**Discussion by sixsox** [Kunzisoft/KeePassDX#1381] [Ideas]
I am missing a global search across all groups and sub-groups in KeePassDX. 
On the desktop I use global search all the time to find the right entries.

**Comment by J-Jamet** [comment]
You just have to launch the search with the icon containing the magnifying glass and not to check the current group which is represented by a chip.

**Reply by sixsox** [reply]
Unfortunately that does not appear to work here as you suggested. 
Search returns some entries with that search term, but not all entries and it does not find search terms in sub-groups. Still trying to figure out, if it has something to do with group settings. However, as far as I am aware my top-level group is searchable and all sub-groups have searchable set to 'inherited'.

**Reply by J-Jamet** [reply]
> Search returns some entries with that search term, but not all entries and it does not find search terms in sub-groups.

I just retested and I can't reproduce your bug. Check that your database is correctly formatted. If you can create a database without sensitive information to reproduce the problem and upload it, it will help identify the issue.

**Reply by sixsox** [reply]
I tried creating a new kdbx file to reproduce the bug - with no luck. 
I can confirm though that global search does work as expected on my kdbx file both with KeePassXC and original Keepass Pro, so the unexpected behavior is exclusive to KeePassDX.

**Reply by J-Jamet** [reply]
If you can't reproduce the problem with another database, I would say that the problem is inherent to your database. That's why I asked if it was well formatted, I want to improve the code to bypass formatting errors in KeePassDX but I have to reproduce it already to debug and make the improvement.

**Reply by ghost** [reply]
What attribute are you searching for? Never had this issue.

**Reply by sixsox** [reply]
I usually search for something in the title field, and yes I have made sure that 'Title' is selected in search settings.
I have checked again that all groups are searchable, but even when I'm within the group of the item I am searching for, the item does not show up in search.
The file in question was created years ago with original KeePass 2.x, now I use KeePassXC and have changed the encryption from AES/AES to ChaCha20/Argon2d. This issue will probably have something to do specifically with this file of mine, however global search works both with KeePass 2.x and KeePassXC on this file, whereas KeePassDX misses some items in global search.
Sadly, I haven't been able to reproduce this with a newly created file.
